tools/ci_connector_ops/ci_connector_ops/pipelines/models.py

```diff
@@ -0,0 +1,99 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+import json
+from dataclasses import dataclass, field
+from datetime import datetime
+from enum import Enum
+from typing import List, Optional, Union
+
+from ci_connector_ops.pipelines.contexts import ConnectorTestContext
+from dagger import Client, Container
+
+
+class Step(Enum):
+    CODE_FORMAT_CHECKS = "Code format checks"
+    PACKAGE_INSTALL = "Package install"
+    UNIT_TESTS = "Unit tests"
+    INTEGRATION_TESTS = "Integration tests"
+    DOCKER_BUILD = "Docker Build"
+    ACCEPTANCE_TESTS = "Acceptance tests"
+    QA_CHECKS = "QA Checks"
+
+    def get_dagger_pipeline(self, dagger_client_or_container: Union[Client, Container]) -> Union[Client, Container]:
+        return dagger_client_or_container.pipeline(self.value)
```

Comment:
Can you explain how a client and a container differ? and why they both have a pipeline?

Reply:
A client is client of Dagger API. 
You can create containers from it, access the host etc... 
A container is a dagger container. 

The Dagger SDK allows creating "pipelines" from clients and containers. 
A pipeline groups client or container in execution in a logical way, it has no impact on the underlying execution flow but will help build the DAG representation and logging structure. 

Calling `pipeline` on a container returns a container. Calling pipeline on a client returns a client.
You would call `pipeline` on the original client if you want to create a new client for a specific group of step.
You would call `pipeline` on a base container if you want to make the next interaction on this container in a specific group of step.

I created this `get_dagger_pipeline` function to make the `Step` model responsible for generating their pipelines.